### DISCUSSION TITLE
When can we expect the refiner to return

### DISCUSSION THREAD

**Discussion by SuperChevyChevy** [lllyasviel/stable-diffusion-webui-forge#1209] [General]
The refiner switch being under maintenance is killing me right now. I've been looking for info but has there been any indication on when/if it will return in the new forge???? Are we expecting it to come back after all the flux fixes and balancing???
I've had to go back to an old, admittedly worse commit, just for this feature. And before i get the "you don't need it" remarks, i need it for work I'm doing for a client and they demand the refiner switch be used.

**Comment by InternalMegaT** [comment]
The disregard is appalling. I'm not even looking to use it for Flux; I'm looking to use it for SDXL, which has already been supported since before Forge existed. Please make the refiner work now. Everyone who has asked that question has been completely ignored. The devs have just been ignoring everyone with that question. There is no reason why they wouldn't have the refiner working when it clearly worked just fine on Automatic1111 especially after nine months. I understand it not working with Flux models, but I don't want it for Flux models; I want it for SD models. I need the refiner to be able to switch between different models during the generation process. This would allow me to achieve more nuanced results that fit within my specific creative vision, be it style and/or composition.

**Reply by ScionoicS** [reply]
You're owed nothing. Wait for their volunteer efforts to come to fruition and be grateful. 

There's a history of entitled people and troll brigades on this project. Go back home because this tone is likely why silence and getting the work done is preferred over engaging with the public. 

It's an open source project. You can always submit a pull request.

**Reply by InternalMegaT** [reply]
I understand the "you're not entitled to free work" argument, this is about basic communication in an open source project where the refiner feature, which worked perfectly in A1111, remains conspicuously absent. You cannot blame me for hating this; this has been an issue sense the beginning of this project. The frustration stems from the complete silence about its development status, even after many users asking the same question. I am simply asking for a feature that many users depend on for their creative work.

**Reply by ArnorWing** [reply]
Refiner works in the February/March version before the updates stopped. 
I still only use this version. It doesn't do FLUX but i don't need flux yet.
This version is rock solid and i can run it for days non stop.

**Reply by realdreamachine** [reply]
I used it a lot too. Generating with SD1.5 then refining some latter steps with RealisticVision. But when the refiner became unavailable, really there's no significant difference when you send from Txt2img to Img2img, swap checkpoints and then upscale. Adjust your settings accordingly and you'll still see excellent results.  

So.. sorry but, _you don't need it_. For now. Tell the client to take a hike and show a bit of gratitude for what we're all enjoying. I mean really, frustration is understandable when one can't get something working right, I know the feeling, but I'm grateful to the developers of this for providing such a neat tool.

Now send your SD1.5 generated picture to Img2img and see what you can do. RealisticVision Versions 5 and 6 are indispensible and I just started using STOIQONewrealityFLUXSD_XL40.safetensors with results nothing short of miraculous. So calm yourself and get on with it.

**Reply by InternalMegaT** [reply]
I use the refiner not for image upscaling, but to fine-tune specific art styles that the base model struggles to generate accurately. For my needs, the refiner is required. I've found that SD 1.5 produces lower quality results, which is why I exclusively use SDXL, SD3, and Flux. The only scenario where SD 1.5 proves useful is when using img2img to process SD 1.5 images through an SDXL model. Otherwise, the output tends to be low-quality with deformities. I need the refiner because base models alone can't handle certain styles I'm trying to create. It's not about switching between models or upscaling - I need the refiner to control exactly how the art looks. Just switching between txt2img and img2img with different models doesn't give me the quality I need. The refiner is a special tool that helps me keep control over the art style. The refiner works by generating part of the image (like 50-80%) with the base model, then seamlessly continues the generation process with a different model to complete the remaining percentage. With img2img, you're taking a fully completed image from one model and essentially regenerating it through another model. Refiner is a one of a kind tool that doesn't have any replacements, again I do need this tool for my generations.

**Reply by realdreamachine** [reply]
I know what it does. You're using two different checkpoints, right? If so, just try doing what I told you.

**Reply by halfblack** [reply]
If you knew and had the ability to properly read, you'd know your answer is inferior at best, and often even useless for the mentioned case. @InternalMegaT - consider SwarmUI for the time being.

**Reply by swamp-cabbage** [reply]
@InternalMegaT  - you should definitely ask for a refund.

**Comment by R4INN** [comment]
I think the best thing to add would just being able to select a different model for hi-res fix.

**Reply by ArnorWing** [reply]
> I think the best thing to add would just being able to select a different model for hi-res fix.

You can already
SETTINGS>UI ALTERNATIVES>Hires fix: show hires checkpoint and sampler selection

**Reply by moremusc** [reply]
> > I think the best thing to add would just being able to select a different model for hi-res fix.
> 
> You can already SETTINGS>UI ALTERNATIVES>Hires fix: show hires checkpoint and sampler selection

This doesn't work either

**Comment by ahblahblah** [comment]
Was it ever fixed? I can't get it to work

**Comment by Seedmanc** [comment]
I'm not even sure it works properly now, no more error, but the image just doesn't feel improved by it, it's slightly different and that's all. I think the refiner is meant to do something more than merely switching models at some timestep, since there are settings about aesthetic score. Currently it seems to ignore them.

**Comment by jt-michels** [comment]
Just to quell the noise -- Anything in Forge that you ask to swap models while it's still in latent will fail. Usually silently (no thrown error) . This Repo has been all but dead for a long time. I'm not sure what the best replacement is, tbh. I still use Forge for ease of use on many generations, but anything advanced I use Comfy.
